forest/tree/tree.go

```diff
@@ -0,0 +1,56 @@
+package tree
+
+import (
+	// cache "github.com/treeverse/lakefs/forest/cache_map"
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/graveler/committed/sstable"
+)
+
+type TreePart struct {
+	PartName sstable.ID   `json:"part_name"`
+	MaxKey   graveler.Key `json:"max_path"`
+}
+type TreeSlice struct {
+	TreeSlice []TreePart
+}
+type DummyMap map[string]string // place holder for cache_map package that will be merged soon
+
+type TreesRepo struct {
+	TreesMap   DummyMap
+	PartManger sstable.Manager
+}
+
+// InitTreeRepository creates the tree cache, and stores part Manager for operations of parts (currently implemented as sstables).
+// should be called at process init.
+// decisions on who calls it and how to get a treesRepository will be taken later
+type InitTreesRepository func(manager sstable.Manager) TreeRepo
+
+type TreeRepo interface {
+	// NewTreeWriter returns a writer that uses the part manager to create a new tree
+	NewTreeWriter(splitFactor int, // average number of keys that we want to stored in a part
+		// for more detail, look at "IsSplitKey"
+		closeAsync sstable.BatchWriterCloser, // component used to close part asynchronously, and wait for all part
```

Comment:
You're braking the abstraction layers, why does the caller of the TreeManager need to pass a sstable writer feature?